src/main/java/app/sami/languageWeb/config/JwtService.java

```diff
@@ -0,0 +1,62 @@
+package app.sami.languageWeb.config;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+
+@Service
+public class JwtService {
+
+    private static final String SECRET_KEY = "2D4B6150645367566B597033733676397924423F4528482B4D6251655468576D";
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public String generateToken(UserDetails userDetails){
+        return generateToken(new HashMap<>(), userDetails);
+    }
+    public String generateToken(Map<String, Object> extraClaims, UserDetails userDetails){
```

Comment:
are all these `generateToken` actually used ?

Reply:
if so, flip the arguments so the signature is always `generateToken(UserDetails, extra)`